lib/ConnectionPool.h

```diff
@@ -41,7 +41,8 @@ using ExecutorServiceProviderPtr = std::shared_ptr<ExecutorServiceProvider>;
 class PULSAR_PUBLIC ConnectionPool {
    public:
     ConnectionPool(const ClientConfiguration& conf, ExecutorServiceProviderPtr executorProvider,
-                   const AuthenticationPtr& authentication, bool poolConnections = true);
+                   const AuthenticationPtr& authentication, bool poolConnections,
```

Comment:
Why delete the default value? Set default values can reduce incoming parameters.

Reply:
Because I added a parameter `clientVersion` at the tail. The default parameter not at the tail is meaningless.

Additional, in C++, a default parameter works by generating one more function symbol, it's better not to add a default value just for tests. In the `lib/` subdirectory, the constructor of `ConnectionPool` is only called in `ClientImpl`'s constructor, so we only need one constructor.

Reply:
So, unless there is a special requirement for a long parameter that requires a default value, in other cases, it is better not to set a default value?

Reply:
IMO, the default arguments feature is used when the overloads can be simplified. For example, assuming we'd have two overloads:

```c++
// Overload 1
void f(int x, int y) { /* ... */ }
// Overload 2
void f(int x) { f(x, 0); }
```

Then we can simplify them to:

```c++
void f(int x, int y = 0) { /* ... */ }
```

It would be more clear.

However, in the case here, we don't need any overload as I've explained before:

> In the lib/ subdirectory, the constructor of ConnectionPool is only called in ClientImpl's constructor, so we only need one constructor.

If the overloads are only used for tests, we should not add them.

Reply:
+1 for not using default value here.